src/riak_pb_dt_codec.erl

```diff
@@ -302,9 +310,13 @@ encode_counter_op({decrement, Int}) when is_integer(Int) ->
 -spec decode_set_op(#setop{}) -> set_op().
 decode_set_op(#setop{adds=A, removes=[]}) ->
     {add_all, A};
-decode_set_op(#setop{adds=[], removes=R}) ->
+decode_set_op(#setop{adds=[], removes=_R, is_gset=true}) ->
+    {error, <<"Can't remove from GSet">>};
```

Comment:
This is where it gets confusing to future generations. A gset can't have a removes field at all, that would make no sense.